src/bridge/bridge.c

```diff
@@ -416,14 +416,24 @@ setup_router (CockpitTransport *transport,
 {
   CockpitRouter *router = NULL;
   GList *bridges = NULL;
+  GList *l = NULL;
 
   packages = cockpit_packages_new ();
   if (!privileged_slave)
       bridges = cockpit_packages_get_bridges (packages);
 
-  router = cockpit_router_new (transport, payload_types, bridges);
+  router = cockpit_router_new (transport, payload_types, NULL);
   add_router_channels (router);
 
+  /* Enumerated in reverse, since the last rule is matched first.
+   * This has to happen after add_router_channels as the
+   * packages based bridges should have priority.
+   */
+  for (l = g_list_last (bridges); l != NULL; l = g_list_previous (l))
+    {
+      cockpit_router_add_bridge (router, l->data);
+    }
```

Comment:
Last rule added is matched against first but we want the bridges ordered by priority so we add them in reverse. This is all spelled out in cockpit router. The important change here is that add_router_channels is added before the bridges instead of after as was happening before.